### PR TITLE
Fix preview deployment

### PR DESCRIPTION
**Pull request by lgonczar**
### Change description ###

Limit Vault name to 24 characters, which is the allowed maximum. Also, remove the default product name from terraform.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: infrastructure/variables.tf [hunk 1]
```diff
@@ -1,6 +1,5 @@
 variable "product" {
   type    = "string"
-  default = "draft-store"
```
Comment (rkondratowicz) [inline infrastructure/variables.tf]: is this related?

Comment (lgonczar) [reply infrastructure/variables.tf]: Possibly. After the first commit terraform complained about not being able to find the server farm and as far as I remember that had something to do with setting the product name in code (might be wrong). This default is unnecessary, anyway.

Comment (timja-kainos) [reply infrastructure/variables.tf]: yeah it shouldn't be in here, it comes from the jenkinsfile now

Comment (lgonczar) [reply infrastructure/variables.tf]: I've changed the title of the PR to be more accurate.

File: infrastructure/main.tf [hunk 2]
```diff
@@ -52,6 +52,8 @@ module "api" {
 # region save DB details to Azure Key Vault
 module "key-vault" {
   source              = "git@github.com:hmcts/moj-module-key-vault?ref=master"
+  # Making sure Vault name will not exceed the maximum length of 24 charactes
+  name                = "${substr(format("%s-%s", var.product, var.env), 0, 24)}"
```
Comment (rkondratowicz) [inline infrastructure/main.tf]: what's the origin of this max length? can we not just increase it?

Comment (timja-kainos) [reply infrastructure/main.tf]: netbios I believe ~30 years ago microsoft limitation ;)

**Comment by lgonczar** [comment]
There's an inconsistency between Terraform state file and the actual state of the infrastructure - closing.
